packages/xrpl/src/sugar/submit.ts

```diff
@@ -153,7 +153,8 @@ export async function waitForFinalTransactionOutcome<
     })
 
   if (txResponse.result.validated) {
-    return txResponse
+    // eslint-disable-next-line @typescript-eslint/consistent-type-assertions -- we know that txResponse is of type TxResponse
+    return txResponse as TxResponse<T>
```

Comment:
Can you leave a comment to fix this later --- the typing is really messy

Reply:
Done